compiler/rustc_session/src/session.rs

```diff
@@ -784,6 +784,13 @@ impl Session {
         }
     }
 
+    pub fn inline_asm_dialect(&self) -> rustc_ast::LlvmAsmDialect {
+        match self.asm_arch {
+            Some(InlineAsmArch::X86 | InlineAsmArch::X86_64) => rustc_ast::LlvmAsmDialect::Intel,
+            _ => rustc_ast::LlvmAsmDialect::Att,
+        }
+    }
```

Comment:
The concept of an asm dialect doesn't exist outside of x86. In fact the `.att_syntax` and `.intel_syntax` directives are x86-specific and don't exist on other targets.